src/fsharp/FSharp.Core/iseq.fs

```diff
@@ -0,0 +1,1971 @@
+                            match lst with
+                            | hd :: tl ->
+                                consumer.ProcessNext hd |> ignore
+                                if outOfBand.HaltedIdx = 0 then
+                                    iterate tl
+                            | _ -> ()
+                        iterate alist
+
+            [<Struct;NoComparison;NoEquality>]
+            type unfold<'S,'T> (generator:'S->option<'T*'S>, state:'S) =
+                interface IIterate<'T> with
+                    member __.Iterate (outOfBand:Folder<'U,'Result,'State>) (consumer:Activity<'T,'U>) =
+                        let generator = generator
+                        let rec iterate current =
+                            match generator current with
+                            | Some (item, next) ->
+                                consumer.ProcessNext item |> ignore
+                                if outOfBand.HaltedIdx = 0 then
+                                    iterate next
+                            | _ -> ()
+                        iterate state
+
+            [<Struct;NoComparison;NoEquality>]
+            type init<'T> (f:int->'T, terminatingIdx:int) =
+                interface IIterate<'T> with
+                    member __.Iterate (outOfBand:Folder<'U,'Result,'State>) (consumer:Activity<'T,'U>) =
+                        let terminatingIdx = terminatingIdx
+                        let f = f
+
+                        let firstIdx = 
+                            match box consumer with
+                            | :? ISkipable as skipping ->
+                                let rec skip idx =
+                                    if idx = terminatingIdx || outOfBand.HaltedIdx <> 0 then
+                                        terminatingIdx
+                                    elif skipping.CanSkip () then
+                                        skip (idx+1)
+                                    else
+                                        idx
+                                skip -1
+                            | _ -> -1
+
+                        let rec iterate idx =
+                            if idx < terminatingIdx then
+                                consumer.ProcessNext (f (idx+1)) |> ignore
+                                if outOfBand.HaltedIdx = 0 then
+                                    iterate (idx+1)
+                                else
+                                    idx
+                            else
+                                idx
+
+                        let finalIdx = iterate firstIdx
+                        if outOfBand.HaltedIdx = 0 && finalIdx = System.Int32.MaxValue then
+                            raise <| System.InvalidOperationException (SR.GetString(SR.enumerationPastIntMaxValue))
+
+            let execute (createFolder:PipeIdx->Folder<'U,'Result,'State>) (transformFactory:TransformFactory<'T,'U>) pipeIdx (executeOn:#IIterate<'T>) =
```

Comment:
Please add `///` comments on many of these - but particularly ``execute`` and ``executeThin`` to explain the difference